numpy/core/tests/test_multiarray.py

```diff
@@ -6723,7 +6723,7 @@ def __rmatmul__(self, other):
     def test_matmul_raises(self):
         assert_raises(TypeError, self.matmul, np.int8(5), np.int8(5))
         assert_raises(TypeError, self.matmul, np.void(b'abc'), np.void(b'abc'))
-        assert_raises(ValueError, self.matmul, np.arange(10), np.void(b'abc'))
+        assert_raises(TypeError, self.matmul, np.arange(10), np.void(b'abc'))
```

Comment:
Does this user-facing change require a release note?

Reply:
This just changes the order of errors, so no. If you do `np.matmul(np.arange(10), np.void([b'abc'] * 10))` you already get the type error right now.